apps/prairielearn/python/zygote.py

```diff
@@ -125,16 +127,15 @@ def __init__(self, seed: int) -> None:
     def find_spec(
         self,
         fullname: str,
-        path: Sequence[str] | None,
-        target: types.ModuleType | None = None,
+        _path: Sequence[str] | None,
+        _target: types.ModuleType | None = None,
     ):
         if fullname == "faker" or fullname.startswith("faker."):
             # Once this initialization is done we no longer need this meta path finder
             sys.meta_path.remove(self)
             from faker import Faker
 
             Faker.seed(self.seed)
-        return None
```

Comment:
I prefer the explicit `return None` here, given this function is [explicitly expected to return a spec or None](https://docs.python.org/3/library/importlib.html#importlib.abc.MetaPathFinder.find_spec), and None has a meaning ("spec cannot be found", and finder should proceed with the next path finder). I know `return` is equivalent, but from a readability point of view `return None` is clearer. I tested, and adding an explicit return type to the method (` -> ModuleSpec | None:`) would cause ruff to no longer complain about return None.

That said, I'm ok if you would like to go in that direction, provided you add a comment that `return None` is implicit and on purpose. Same for the similar change a few lines above this one.

Reply:
So these don't actually return anything AFAIK. Perhaps a rename to `validate_spec` would make it more obvious that it doesn't return?

Reply:
This meta finder is added to the zygote so that, as soon as an "import faker" is found in the code, the faker library's seed is initialized, but if the code doesn't use faker then faker is not initialized. The method implements a specification of the superclass, and is called to identify the module spec to return, but we only use it as a means to find out if faker is used, so we explicitly return None so that the regular finder takes over.

Reply:
Going to resolve this, and add a link to a more relevant PR.